src/sql/workbench/parts/dashboard/browser/widgets/explorer/explorerTreeContext.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable } from 'vs/base/common/lifecycle';
+import { IContextKey, IContextKeyService, RawContextKey } from 'vs/platform/contextkey/common/contextkey';
+import { IConnectionProfile } from 'sql/platform/connection/common/interfaces';
+import { MetadataType } from 'sql/platform/connection/common/connectionManagement';
+import { ObjectMetadataWrapper } from 'sql/workbench/parts/dashboard/browser/widgets/explorer/objectMetadataWrapper';
+
+export declare type ContextResource = IConnectionProfile | ObjectMetadataWrapper;
+
+export interface IContextValue {
+	resource: ContextResource;
+	providerName: string;
+	isCloud: boolean;
+}
+
+export class ItemContextKey extends Disposable implements IContextKey<IContextValue> {
+
+	static readonly ItemType = new RawContextKey<string>('itemType', undefined);
+	static readonly Item = new RawContextKey<IContextValue>('item', undefined);
+	static readonly ConnectionProvider = new RawContextKey<string>('provider', undefined);
+	static readonly IsCloud = new RawContextKey<boolean>('isCloud', undefined);
+
+	private _itemTypeKey: IContextKey<string>;
+	private _itemKey: IContextKey<IContextValue>;
+	private _connectionProviderKey: IContextKey<string>;
+	private _isCloudKey: IContextKey<boolean>;
+
+	constructor(
+		@IContextKeyService contextKeyService: IContextKeyService
+	) {
+		super();
+
+		this._itemTypeKey = ItemContextKey.ItemType.bindTo(contextKeyService);
+		this._itemKey = ItemContextKey.Item.bindTo(contextKeyService);
+		this._connectionProviderKey = ItemContextKey.ConnectionProvider.bindTo(contextKeyService);
+		this._isCloudKey = ItemContextKey.IsCloud.bindTo(contextKeyService);
+	}
+
+	set(value: IContextValue) {
+		this._itemKey.set(value);
+		this._connectionProviderKey.set(value.providerName.toLowerCase());
+		this._isCloudKey.set(value.isCloud);
+		if (value.resource instanceof ObjectMetadataWrapper) {
+			switch (value.resource.metadataType) {
+				case MetadataType.Function:
+					this._itemTypeKey.set('function');
+					break;
```

Comment:
If I check in my PR for fixing the item metadata first make sure this is fixed too - there's a couple new types added and the default should be Unknown instead (why do we even have the default be database for anything that doesn't have metadata...that seems like it's assuming a whole lot)